tuna/go_fish.py

```diff
@@ -40,29 +41,56 @@ def parse_args():
   parser = argparse.ArgumentParser()
 
   parser.add_argument('lib',
+                      nargs='?',
+                      default=Library.MIOPEN,
                       type=Library,
                       help="Specify library to run",
                       choices=Library)
 
+  parser.add_argument('--yaml',
+                      dest='yaml',
+                      default=None,
+                      help='Path to yaml input file')
+
   args, _ = parser.parse_known_args()
   return vars(args)
 
 
 def main():
   """Main function to start Tuna"""
   LOGGER.info(sys.argv)
+  LOGGER.info(len(sys.argv))
   args = parse_args()
-  library = get_library(args)
+  if '--yaml' in sys.argv and len(sys.argv) > 4:
+    LOGGER.error('Command like arguments not accepted with yaml file')
+    return False
+
+  #case no yaml file
+  libraries = []
+  libraries.append(get_library(args))
+  yaml_files = [args['yaml']]
+
+  #case with yaml file
+  if args['yaml']:
+    yaml_files = parse_yaml(args['yaml'], args['lib'])
```

Comment:
now yaml_dicts?

Reply:
here they are files, yaml_files here are actual paths to /tmp/<random_name>.yaml